UM/Qt/qml/UM/MessageStack.qml

```diff
@@ -91,16 +92,122 @@ ListView
+                }
+                UM.RecolorImage
+                {
+                    id: messageIcon
+                    height: parent.height
+                    width: parent.width
+                    sourceSize.width: width
+                    sourceSize.height: height
+                }
+
+                states:
+                [
+                    State
+                    {
+                        name: "positive"
+                        when: model.message_type == 0
+                        PropertyChanges
+                        {
+                            target: messageIcon
+                            source: UM.Theme.getIcon("Check", "low")
+                            color: UM.Theme.getColor("message_success_icon")
+                        }
+                        PropertyChanges
+                        {
+                            target: messageIconBackground
+                            color: UM.Theme.getColor("message_success_icon_background")
+                        }
+                    },
+                    State
+                    {
+                        name: "neutral"
+                        when: model.message_type == 1
+                        PropertyChanges
+                        {
+                            target: messageIcon
+                            source: ""
+                        }
+                        PropertyChanges
+                        {
+                            target: messageIconBackground
+                            color: "transparent"
+                        }
+                    },
+                    State
+                    {
+                        name: "warning"
+                        when: model.message_type == 2
+                        PropertyChanges
+                        {
+                            target: messageIcon
+                            source: UM.Theme.getIcon("Warning", "low")
+                            color: UM.Theme.getColor("message_warning_icon")
+                        }
+                        PropertyChanges
+                        {
+                            target: messageIconBackground
+                            color: UM.Theme.getColor("message_warning_icon_background")
+                        }
+                    },
+                    State
+                    {
+                        name: "error"
+                        when: model.message_type == 3
+                        PropertyChanges
+                        {
+                            target: messageIcon
+                            source: UM.Theme.getIcon("Cancel", "low")
+                            color: UM.Theme.getColor("message_error_icon")
+                        }
+                        PropertyChanges
+                        {
+                            target: messageIconBackground
+                            color: UM.Theme.getColor("message_error_icon_background")
```

Comment:
Rather than hard-coding specific theme entries for specific details in the interface, the design philosophy is to have re-usable theme entries as much as possible.
A better theme entry could be something like `error_contrast` or `error_background`. Although in this case it appears to be more like a foreground than a background.